pkg/apis/nmstate/v1alpha1/nodenetworkconfigurationpolicy_types_test.go

```diff
@@ -0,0 +1,135 @@
+package v1alpha1
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+var _ = Describe("Node info list", func() {
+	Context("is empty", func() {
+		originalNodeInfoList := NodeInfoList{}
+		var newNodeInfoList NodeInfoList
+
+		BeforeEach(func() {
+			newNodeInfoList = originalNodeInfoList.DeepCopy()
+		})
+
+		It("should return nil when finding a condition", func() {
+			foundCondition := newNodeInfoList.FindCondition("foo_node", ConditionType("bar"))
+			Expect(foundCondition).To(BeNil())
+		})
+
+		Context("and we set a new condition", func() {
+			addedNodeName := "shruberry"
+			addedConditionType := ConditionType("foo")
+			addedConditionStatus := corev1.ConditionTrue
+			addedConditionReason := ConditionReason("bar")
+			addedConditionMessage := "baz"
+
+			BeforeEach(func() {
+				newNodeInfoList.SetCondition(addedNodeName, addedConditionType, addedConditionStatus, addedConditionReason, addedConditionMessage)
+			})
+
+			It("should extend the list", func() {
+				Expect(newNodeInfoList).To(HaveLen(len(originalNodeInfoList) + 1))
```

Comment:
len(1) originalNodeInfoList is overkill here.

Reply:
Read about this on some BDD rant/tutorial. We should not assume that the original length was 0, that is not what we are testing anyway. We are testing that the list will be extended, no matter the original length.